vec3d.h

```diff
@@ -0,0 +1,130 @@
+#include<math.h>
+
+
+template<typename T>
+class Vec3{
+    private:
+        T x, y, z;
+    public:
+        Vec3() : x(T(0)), y(T(0)), z(T(0)) {}           
+        Vec3(T xx) : x(xx), y(xx), z(xx) {}
+        Vec3(T xx, T yy, T zz): x(xx), y(yy), z(zz) {}
+
+        T getx(){return x;}
+        T gety(){return y;}
+        T getz(){return z;}
+        void xin(T &xx){x = xx;}
+        void yin(T &yy){y = yy;}
+        void zin(T &zz){z = zz;}
+
+
+
+
+        Vec3<T> operator * (const T &f){
+            return Vec3<T>(x*f, y*f, z*f);
+        }
+        Vec3<T> operator * (const Vec3<T> &v){
+            return Vec3<T>(x*v.x, y * v.y, z * v.z);
+        }
+
+
+        Vec3<T> operator - (const Vec3<T> &v){
+            return Vec3<T>(x - v.x, y - v.y, z - v.z);
+        }
+        Vec3<T> operator + (const Vec3<T> &v){
+            return Vec3<T>(x + v.x, y + v.y, z + v.z);
+        }
+
+        Vec3<T>& operator += (const Vec3<T> &v){
+            x += v.x, y += v.y, z += v.z;
+            return *this;
+        }
+
+        Vec3<T>& operator *= (const Vec3<T> &v){
+            x *= v.x, y *= v.y, z *= v.z;
+            return *this;
+        }
+
+        Vec3<T> operator -(){
+            return Vec3<T>(-x, -y, -z);
+        }
+
+        void operator = (const Vec3<T> &v){
+            this->x = v.x;
+            this->y = v.y;
+            this->z = v.z;
+        }
+
+
+        
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+        T length_sq(){
+            return x*x + y*y + z*z;
+        }
+        T length(){
+            return sqrt(length_sq());
+        }
+        T dot(const Vec3<T> &v){
+            return x * v.x + y * v.y + z * v.z;
+        }
+        Vec3<T> unit(){
+            T l = length();
+            return Vec3<T>(x/l, y/l, z/l);
+        }
+
+        void convt_unit(){
```

Comment:
I don't think `convt_unit` is a very good name. At the very list, it should be something like `convt_to_unit`, but something like `convertToUnitVector` is more in line with our function naming scheme. Or, just `unitVector` works too.

Reply:
Is this function really required, even though you already have a `unit()` function?